### PR TITLE
Report failing filenames when --fix=no in mixed-line-endings

### PR DESCRIPTION
**Pull request by gimbo**
Fixes [issue 340](https://github.com/pre-commit/pre-commit-hooks/issues/340).

### CODE REVIEW

File: pre_commit_hooks/mixed_line_ending.py [hunk 1]
```diff
@@ -37,9 +37,15 @@ def fix_filename(filename, fix):
...
-    if fix == 'no' or (fix == 'auto' and not mixed):
+    if fix == 'no':
+        if mixed:
+            # Not fixing, just report problematic file
+            print('Mixed line endings: {filename}'.format(filename=filename))
         return mixed
 
+    if fix == 'auto' and not mixed:
+        return False
+
```
Comment (asottile) [inline pre_commit_hooks/mixed_line_ending.py]: a smaller patch which accomplishes the same thing and keeps the printing in `main()`:

```diff
-retv |= fix_filename(filename, args.fix)
+if fix_filename(filename, args.fix):
+    if args.fix == 'no':
+        print('{}: mixed line endings'.format(filename))
+    else:
+        print('{}: fixed mixed line endings.format(filename))
+    retv = 1
```

**Comment by asottile** [comment]
looks good -- now all that's left is to add a quick test for this and it should be good to merge!

You'll find the [`capsys`](https://docs.pytest.org/en/latest/capture.html#accessing-captured-output-from-a-test-function) fixture useful for testing this behaviour -- if you grep around there's probably a few other examples in this repo

**Comment by asottile** [comment]
@gimbo let me know if you want some help writing the test :)

**Comment by gimbo** [comment]
@asottile Thanks, it's not really a matter of needing help, it's more about time/other commitments. I may get to it; if someone else gets to it before me, great.

**Comment by asottile** [comment]
:tada: thanks for the PR @gimbo -- I added some quick lines to test the new output and it looks good!

**Comment by gimbo** [comment]
Thanks @asottile - that's great! Sorry I wasn't able to close the loop on this one but other work got frantic all of a sudden. :-)

**Pull request merged**